src/EditorFeatures/CSharp/MisplacedUsingDirectives/MisplacedUsingDirectivesCodeFixProvider.cs

```diff
@@ -0,0 +1,394 @@
+                usingDirective => ExpandUsingDirectiveAsync(document, usingDirective, cancellationToken));
+            var newCompilationUnitUsings = await Task.WhenAll(expandCompilationUnitUsingsTasks).ConfigureAwait(false);
+
+            var compilationUnitWithExpandedUsings = compilationUnit.WithUsings(new SyntaxList<UsingDirectiveSyntax>(newCompilationUnitUsings));
+
+            // To expand the using declarations in the namespace we need to get an updated document, since the 
+            // expander requires that the SyntaxTree for the semantic model match that of the using directives.
+            var newDocument = document.WithSyntaxRoot(compilationUnitWithExpandedUsings);
+            var newCompilationUnit = (CompilationUnitSyntax)await newDocument.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            // Expand the using declarations in the namespace declaration.
+            var namespaceDeclaration = newCompilationUnit.Members.OfType<NamespaceDeclarationSyntax>().First();
+
+            var expandNamespaceDeclarationUsingsTasks = namespaceDeclaration.Usings.Select(
+                usingDirective => ExpandUsingDirectiveAsync(newDocument, usingDirective, cancellationToken));
+            var newNamespaceDeclarationUsings = await Task.WhenAll(expandNamespaceDeclarationUsingsTasks).ConfigureAwait(false);
+
+            var namespaceDeclarationWithExpandedUsings = namespaceDeclaration.WithUsings(new SyntaxList<UsingDirectiveSyntax>(newNamespaceDeclarationUsings));
+
+            return newCompilationUnit.ReplaceNode(namespaceDeclaration, namespaceDeclarationWithExpandedUsings);
+        }
+
+        private static async Task<UsingDirectiveSyntax> ExpandUsingDirectiveAsync(Document document, UsingDirectiveSyntax usingDirective, CancellationToken cancellationToken)
+        {
+            var newName = await Simplifier.ExpandAsync(usingDirective.Name, document, cancellationToken: cancellationToken).ConfigureAwait(false);
+            return usingDirective.WithName(newName);
+        }
+
+        private static CompilationUnitSyntax MoveUsingsInsideNamespace(CompilationUnitSyntax compilationUnit)
+        {
+            // Get the compilation unit usings and set them up to format when moved.
+            var usingsToAdd = compilationUnit.Usings.Select(
+                directive => directive.WithAdditionalAnnotations(Formatter.Annotation));
+
+            // Remove usings and fix leading trivia for compilation unit.
+            var compilationUnitWithoutUsings = compilationUnit.WithUsings(new SyntaxList<UsingDirectiveSyntax>());
+            var compilationUnitWithoutBlankLine = RemoveLeadingBlankLinesFromFirstMember(compilationUnitWithoutUsings);
+
+            // Fix the leading trivia for the namespace declaration.
+            var namespaceDeclaration = compilationUnitWithoutBlankLine.Members.OfType<NamespaceDeclarationSyntax>().First();
```

Comment:
don't you know there's only one namespace?  so this could just be `.Members[0]` right?  i'd almost prefer that sinc ethis code makes it seem as if you expect multiple namespaces